buildSrc/src/main/java/com/r3/csde/VNodesHelper.java

```diff
@@ -0,0 +1,256 @@
+
+            if (matches.size() == 0) {
+                throw new CsdeException("Registration failed because virtual node for '" + vn.getX500Name() + "' not found.");
+            } else if (matches.size() >1 ) {
+                throw new CsdeException(("Registration failed because more than virtual node for '" + vn.getX500Name() + "'"));
+            }
+
+            String shortHash = matches.get(0).getHoldingIdentity().getShortHash();
+
+            if (!checkVNodeIsRegistered(shortHash)){
+                registerVnode(vn, shortHash);
+            }
+        }
+    }
+    private void registerVnode( VNode vn, String shortHash) throws CsdeException {
+
+        pc.out.println("Registering vNode: " + vn.getX500Name() + " with shortHash: " + shortHash);
+
+        // Configure the registration body (notary vs non notary)
+        String registrationBody;
+        if (vn.getServiceX500Name() == null) {
+            registrationBody =
+                    "{ \"action\" : \"requestJoin\"," +
+                            " \"context\" : {" +
+                            " \"corda.key.scheme\" : \"CORDA.ECDSA.SECP256R1\" } }";
+        } else {
+            registrationBody =
+                    "{ \"action\" : \"requestJoin\"," +
+                            " \"context\" : {" +
+                            " \"corda.key.scheme\" : \"CORDA.ECDSA.SECP256R1\", " +
+                            " \"corda.roles.0\" : \"notary\", " +
+                            " \"corda.notary.service.name\" : \"" + vn.getServiceX500Name() + "\", " +
+                            " \"corda.notary.service.plugin\" : \"net.corda.notary.NonValidatingNotary\" } }";
+        }
+
+        HttpResponse<JsonNode> response = Unirest.post(pc.baseURL + "/api/v1/membership/" + shortHash)
+                .body(registrationBody)
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if (response.getStatus() == HTTP_OK) {
+            pc.out.println("Membership requested for node " + shortHash);
+        } else {
+            throw new CsdeException("Failed to register virtual node "+ shortHash +
+                    ", response status: " + response.getStatus() );
+            }
+
+        // wait until Vnode registered
+        pollForRegistration(shortHash, 30000, 1000);
+
+    }
+
+    // Checks if a virtual node with given shortHash has been registered
+    private boolean checkVNodeIsRegistered(String shortHash) throws CsdeException {
+
+        // Queries registration status for vnode.
+        HttpResponse<JsonNode> response = Unirest.get(pc.baseURL + "/api/v1/membership/" + shortHash)
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if(response.getStatus() != HTTP_OK)
+            throw new CsdeException("Failed to check registration status for virtual node '" + shortHash +
+                    "' response status: " + response.getStatus());
+
+        try {
```

Comment:
Is the indentation in the try/catch flat on purpose?

Reply:
nope